node/internal.ts

```diff
@@ -426,7 +427,7 @@ export function _which(tool: string, check?: boolean): string {
 function _tryGetExecutablePath(filePath: string, extensions: string[]): string {
     try {
         // test file exists
-        let stats: fs.Stats = fs.statSync(filePath);
+        let stats: fs.Stats = fs.statSync(_unTildify(filePath));
```

Comment:
If someone calls `tl.which('sometool')` and it returns a tildified-file-path, won't that cause problem depending on what the path is then used for? For instance, if they stat or lstat it will blow up. I would imagine other things may error as well.

I think it would make more sense to do this in the which function when iterating over the directories. And probably only when home is truthy and process.platform != 'win32' and the dir is `~` or starts with `~/`. Also I noticed the regex below also handles `~\` which doesn't seem correct to me.

Also I'm curious if you have insight into how common this problem is? The stance node took is interesting, and I think makes sense for node. However, since the task lib exposes a which() function, it makes me feel like we have a better argument to add it (pragmatic reasoning to avoid common confusion).